[miopen-hip] can't build miopen

Hi guys,

Can someone build miopen from AUR? It is not available from arch4edu.
I get the following error:

```
yay -S miopen        
:: Checking for conflicts...
:: Checking for inner conflicts...
[Aur: 1]  miopen-3.5.0-1

  1 miopen                                   (Build Files Exist)
==> Packages to cleanBuild?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> A
:: Deleting (1/1): /home/fernando/.cache/yay/miopen
:: Downloaded PKGBUILD (1/1): miopen
  1 miopen                                   (Build Files Exist)
==> Diffs to show?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> 
:: (1/1) Parsing SRCINFO: miopen
==> Making package: miopen 3.5.0-1 (Tue Jun 23 19:02:04 2020)
==> Retrieving sources...
  -> Downloading miopen-3.5.0...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   139  100   139    0     0    220      0 --:--:-- --:--:-- --:--:--   220
100 23.3M    0 23.3M    0     0  3455k      0 --:--:--  0:00:06 --:--:-- 4155k
==> Validating source files with sha256sums...
    miopen-3.5.0 ... Passed
==> Making package: miopen 3.5.0-1 (Tue Jun 23 19:02:11 2020)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found miopen-3.5.0
==> Validating source files with sha256sums...
    miopen-3.5.0 ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting miopen-3.5.0 with bsdtar
==> Sources are ready.
==> Making package: miopen 3.5.0-1 (Tue Jun 23 19:02:13 2020)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting build()...
-- The C compiler identification is GNU 10.1.0
-- The CXX compiler identification is GNU 10.1.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.7.3") 
-- Checking for module 'sqlite3'
--   Found sqlite3, version 3.32.3
-- Found BZip2: /usr/lib/libbz2.so (found version "1.0.8") 
-- Looking for BZ2_bzCompressInit
-- Looking for BZ2_bzCompressInit - found
-- Performing Test HAS_HIP
-- Performing Test HAS_HIP - Failed
-- Hip compiler flags:  -x hip --hip-device-lib-path=/opt/rocm/lib   -D__HIP_ROCclr__=1 -isystem /opt/rocm/hip/../include -isystem /opt/rocm/llvm/lib/clang/11.0.0/include/..  -D__HIP_PLATFORM_HCC__=1  -D__HIP_ROCclr__=1 -isystem /opt/rocm/hip/include -isystem /opt/rocm/include --hip-device-lib-path=/opt/rocm/lib --hip-link  
-- hip compiler: /opt/rocm/bin/clang-ocl
-- Build with rocblas
-- Build without scgemm
-- HIP backend selected.
-- clang-offload-bundler not found
-- extractkernel found: /opt/rocm/hip/bin/extractkernel
-- AMDGCN assembler: /opt/rocm/llvm/bin/clang
-- Build without miopengemm
-- Found Boost: /usr/include (found version "1.72.0") found components: filesystem system 
-- Clang tidy found: 10.0.0
-- Clang tidy checks: *,-android-cloexec-fopen,-cert-msc30-c,-bugprone-exception-escape,-cert-env33-c,-cert-msc32-c,-cert-msc50-cpp,-cert-msc51-cpp,-clang-analyzer-alpha.core.CastToStruct,-clang-analyzer-optin.performance.Padding,-clang-diagnostic-deprecated-declarations,-clang-diagnostic-extern-c-compat,-clang-diagnostic-unused-command-line-argument,-cppcoreguidelines-avoid-c-arrays,-cppcoreguidelines-avoid-magic-numbers,-cppcoreguidelines-explicit-virtual-functions,-cppcoreguidelines-init-variables,-cppcoreguidelines-macro-usage,-cppcoreguidelines-non-private-member-variables-in-classes,-cppcoreguidelines-pro-bounds-array-to-pointer-decay,-cppcoreguidelines-pro-bounds-constant-array-index,-cppcoreguidelines-pro-bounds-pointer-arithmetic,-cppcoreguidelines-pro-type-member-init,-cppcoreguidelines-pro-type-reinterpret-cast,-cppcoreguidelines-pro-type-union-access,-cppcoreguidelines-pro-type-vararg,-cppcoreguidelines-special-member-functions,-fuchsia-*,-google-explicit-constructor,-google-readability-braces-around-statements,-google-readability-todo,-google-runtime-int,-google-runtime-references,-hicpp-braces-around-statements,-hicpp-explicit-conversions,-hicpp-no-array-decay,-hicpp-avoid-c-arrays,-hicpp-signed-bitwise,-hicpp-special-member-functions,-hicpp-uppercase-literal-suffix,-hicpp-use-auto,-hicpp-use-equals-default,-hicpp-use-override,-llvm-header-guard,-llvm-include-order,-misc-misplaced-const,-misc-non-private-member-variables-in-classes,-modernize-avoid-c-arrays,-modernize-pass-by-value,-modernize-use-auto,-modernize-use-default-member-init,-modernize-use-equals-default,-modernize-use-trailing-return-type,-modernize-use-transparent-functors,-performance-unnecessary-value-param,-readability-braces-around-statements,-readability-else-after-return,-readability-isolate-declaration,-readability-magic-numbers,-readability-named-parameter,-readability-uppercase-literal-suffix,-readability-convert-member-functions-to-static
-- Found LATEX: /usr/bin/latex   
-- MIOpen_VERSION= 2.4.0.11-6f333df
-- CMAKE_BUILD_TYPE= Release
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Librt: /usr/lib/librt.so
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Performing Test MIOPEN_HAS_address
-- Performing Test MIOPEN_HAS_address - Failed
-- Performing Test MIOPEN_HAS_thread
-- Performing Test MIOPEN_HAS_thread - Failed
-- Configuring done
-- Generating done
-- Build files have been written to: /home/fernando/.cache/yay/miopen/src/build
Scanning dependencies of target addkernels
[  1%] Building CXX object addkernels/CMakeFiles/addkernels.dir/include_inliner.cpp.o
[  1%] Building CXX object addkernels/CMakeFiles/addkernels.dir/addkernels.cpp.o
[  1%] Linking CXX executable ../bin/addkernels
[  1%] Built target addkernels
[  1%] Inlining MIOpen HIP kernel includes
[  1%] Inlining MIOpen kernels
Scanning dependencies of target MIOpen
[  1%] Building CXX object src/CMakeFiles/MIOpen.dir/buffer_info.cpp.o
c++: error: unrecognized command-line option '--hip-device-lib-path=/opt/rocm/lib'
c++: error: unrecognized command-line option '--cuda-gpu-arch=gfx900'
c++: error: unrecognized command-line option '--cuda-gpu-arch=gfx906'
make[2]: *** [src/CMakeFiles/MIOpen.dir/build.make:430: src/CMakeFiles/MIOpen.dir/buffer_info.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2254: src/CMakeFiles/MIOpen.dir/all] Error 2
make: *** [Makefile:183: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
error making: %!s(func() string=0x55d51bd89ed0)
```

thanks

I tried `miopen-opencl`, it installed without issues from AUR but tensorflow does not detect my AMD GPU with it, but maybe I'm lacking some other package or something else is wrong. Anyway, that is off topic.

Building `miopen-hip` failed, this time later on. https://pastebin.com/rsXEky5y
The paste is kinda huge and I apologize for that, and I guess that you are mostly interested in the final error, but I noticed that somewhere in the middle of the compile process, in line 2778 pf the pastebin, there is this:
```
/home/fernando/.cache/yay/miopen-hip/src/MIOpen-rocm-3.5.0/include/miopen/miopen.h:47:10: error: 'hip/hip_runtime_api.h' file not found [clang-diagnostic-error]
#include <hip/hip_runtime_api.h>
```
I'm not sure if it is relevant or not but I preferred calling your attention to it. There may be other issues in there but I'm not an expert on this.

Again, thank you for your efforts

--- a/miopen-hip/.SRCINFO
+++ b/miopen-hip/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = miopen-hip
 	pkgdesc = AMD's Machine Intelligence Library (HIP backend)
 	pkgver = 3.5.0
-	pkgrel = 3
+	pkgrel = 4
 	url = https://github.com/ROCmSoftwarePlatform/MIOpen
 	arch = x86_64
 	license = custom:NCSAOSL
@@ -9,6 +9,8 @@ pkgbase = miopen-hip
 	makedepends = rocm-cmake
 	makedepends = half
 	makedepends = miopengemm
+	makedepends = clang
+	makedepends = rocminfo
 	depends = rocblas
 	depends = boost
 	depends = llvm-amdgpu

--- a/miopen-hip/PKGBUILD
+++ b/miopen-hip/PKGBUILD
@@ -2,13 +2,13 @@
 
 pkgname=miopen-hip
 pkgver=3.5.0
-pkgrel=3
+pkgrel=4
 pkgdesc="AMD's Machine Intelligence Library (HIP backend)"
 arch=('x86_64')
 url="https://github.com/ROCmSoftwarePlatform/MIOpen"
 license=('custom:NCSAOSL')
 depends=('rocblas' 'boost' 'llvm-amdgpu' 'rocm-clang-ocl' 'hip')
-makedepends=('cmake' 'rocm-cmake' 'half' 'miopengemm')
+makedepends=('cmake' 'rocm-cmake' 'half' 'miopengemm' 'clang' 'rocminfo')
 provides=('miopen')
 conflicts=('miopen')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/rocm-$pkgver.tar.gz")

